modules/src/wake.py

```diff
@@ -0,0 +1,59 @@
+import config
+import modules
+from templates.text import TextTemplate
+
+from datetime import datetime, timedelta
+from time import strptime
+
+def calculate_time(input_time=None): #Input must be in hh:mm format
+    hours = [9, 7, 6, 4] 
```

Comment:
Let's extract the time calculation part (leaving the 'The best time to sleep...' string) out to a separate function and write a few tests on that.